### DISCUSSION TITLE
How do I update to a new version?

### DISCUSSION THREAD

**Discussion by TheChosen7** [AbdBarho/stable-diffusion-webui-docker#421] [Q&A]
Once a release is up. How can I update my existing Stable Diffusion to a new version? 

Trying to press pull within Windows Docker Images does not seem to work. 

(HTTP code 404) unexpected - pull access denied for sd-auto, repository does not exist or may require 'docker login': denied: requested access to the resource is denied

Last time I updated simply by re-downloading/re-building the whole thing in a different folder.

**Comment by AbdBarho** [comment]
just do `git pull` assuming you are using git and you should be good to go.

**Comment by TheChosen7** [comment]
I am not using git , I am using your setup guide which says:

clone this repo and run:
```

docker compose --profile download up --build
#wait until its done, then:
docker compose --profile [ui] up --build
#where [ui] is one of: invoke | auto | auto-cpu | comfy | comfy-cpu | sygil | sygil-sl
```

So I need to re-download the code , replace the one in the existing version and re-run the commands?

**Reply by oddomatik** [reply]
Change directory into the repo you cloned (straight from the guide you quoted), and run a "git pull" from the command line. Or using git utils in windows, or whatever. Then rebuild the image using docker-compose and it will automatically respawn the container with updated versions.

**Reply by oddomatik** [reply]
I just read your error a bit better. Perhaps you are not running the pull in the right directory? You must be in the root of the stable-diffusion-webui-docker directory. It sounds like you are probably in some directory called "sd-auto" when running the program. I'm not familiar with windows utilities but you'll need to point git to the proper directory.

**Reply by AbdBarho** [reply]
I think the original error is because of something like
```
docker pull sd-auto
```
or similar, which fails because the container does not exist on docker-hub.
